### PR TITLE
chore: remove need for removing +234 from numbers to detect

### LINKED ISSUE #49: App should work with or without +234 prepended
**Is your feature request related to a problem? Please describe.**
App allowing numbers without +234

**Describe the solution you'd like**
The app should work with or without +234 prepended

**Describe alternatives you've considered**
Sticking with the existing solution

### PR DESCRIPTION
**Pull request by vicradon**
Fixes #49

### CODE REVIEW

File: src/network-detect.js [hunk 1]
```diff
@@ -89,7 +89,7 @@ NetworkDetect.prototype = {
     }
 
     //Check if number contains unwanted characters
-    if (this.phone.match(/[^0-9]/)) {
+    if (this.phone.slice(1, this.phone.length).match(/[^0-9]/) && this.phone[0] !== '+') {
```
Comment (BolajiAyodeji) [inline src/network-detect.js:92]: With this approach, it doesn't consider the **+** as unwanted if it's at the start of the digits. However, if I put a **+** or any other character somewhere other than position 0, this rule will fail (which we don't want).

Like so: +234819847360+, +234819847&608, +234#198473608 and others.

Also, digits like +2348198473 will also pass since they are 11 digits and start with a **+**.

You need to instead check and ensure it contains just one **+** and at position 0 only for 14 digits, this rule should not work for 11 digits.

**Review by BolajiAyodeji — CHANGES_REQUESTED** [review]
Thanks for this amazing PR chief!

Please take a look at my comment so we can proceed.

File: src/network-detect.js (hunk 1 above)
Comment (vicradon) [reply src/network-detect.js]: @BolajiAyodeji 
I have noted the corrections. Thanks for the review

**Comment by vicradon** [comment]
@BolajiAyodeji 
I have pushed new changes on this branch

File: src/network-detect.js [hunk 2]
```diff
@@ -89,24 +89,33 @@ NetworkDetect.prototype = {
...
 
     //Check if number contains unwanted characters
-    if (this.phone.slice(1, this.phone.length).match(/[^0-9]/) && this.phone[0] !== '+') {
+    if (this.phone[0] === '+' && this.phone.slice(1, this.phone.length).match(/[^0-9]/)) {
+      throw new Error('Number contains unwanted characters')
+    } else if (this.phone[0] !== '+' && this.phone.match(/[^0-9]/)) {
       throw new Error('Number contains unwanted characters')
     }
 
     //Check if number is lesser than 11
```
Comment (BolajiAyodeji) [inline src/network-detect.js:98]: Can you make this _less than_ too?

**Review by BolajiAyodeji — CHANGES_REQUESTED** [review]
Thanks for the new changes, you're amazing!!

The current logic works fine now excepts that it doesn't check if the **+234** prefix is actually valid. Appending a 🇨🇦 **+151** or another other random 3 digits with a + before will still check as a valid number which shouldn't be the case.

_+1518109345321_ will hence check as an MTN number.

**Comment by vicradon** [comment]
Hello Chief @BolajiAyodeji.

I have handled the edge case of non +234 prepends.

Do you have a suite a test cases contributors can check against? Before actual unit tests are written?

**Comment by BolajiAyodeji** [comment]
Hi @vicradon, my sincere apologies for the delayed response. I got caught up with a lot and forgot about this project.

I'll take a look at your recent commits and attend to them. Also, I don't have a suite of test cases contributors can check against for now. Thanks a lot for your patience!

**Review by BolajiAyodeji — APPROVED** [review]
Thanks for the great contribution!

**Comment by BolajiAyodeji** [comment]
@all-contributors please add @vicradon for code

**Comment by vicradon** [comment]
Thank you for merging my PR @BolajiAyodeji.
